### ISSUE TITLE
Unpinned tests fail for workflow

### ISSUE THREAD

**Issue by jpswinski** [ICESat2-SlideRule/sliderule-python#65]
Tests / 3.8 pass
Tests / 3.9 pass
Tests / unpinned-test fails

```
Collecting fiona>=1.8
  Downloading Fiona-1.8.20.tar.gz (1.3 MB)
  Preparing metadata (setup.py): started
  Preparing metadata (setup.py): finished with status 'error'
  ERROR: Command errored out with exit status 1:
   command: /usr/share/miniconda/envs/sliderule-ci/bin/python -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-am2crumu/fiona_400bd1c90cb6458da6565ef3aa233678/setup.py'"'"'; __file__='"'"'/tmp/pip-install-am2crumu/fiona_400bd1c90cb6458da6565ef3aa233678/setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base /tmp/pip-pip-egg-info-mmdt3okl
       cwd: /tmp/pip-install-am2crumu/fiona_400bd1c90cb6458da6565ef3aa233678/
  Complete output (2 lines):
  Failed to get options via gdal-config: [Errno 2] No such file or directory: 'gdal-config'
  A GDAL API version must be specified. Provide a path to gdal-config using a GDAL_CONFIG environment variable or use a GDAL_VERSION environment variable.
```

**Comment by scottyhq** [comment]
The 'unpinned' environment file installs the latest python from conda-forge (3.10) and everything else from pip. For fiona there is no pre-built "wheel" for 3.10, so it is failing on attempting to install from source code (see https://pythonspeed.com/articles/conda-vs-pip/). 

So the CI will continue to fail until the fiona library is updated and released - https://github.com/Toblerity/Fiona/pull/1049. As a workaround until then you can get fiona from conda-forge by adding it here https://github.com/ICESat2-SlideRule/sliderule-python/blob/main/ci/environment-unpinned.yml

**Comment by jpswinski** [comment]
Implemented @scottyhq suggestion above (9b30cde); and now all tests are passing.
